docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-  Type**.
-- On **Arch Linux**, you can easily install CUDA via `pacman -S cuda`
-  without downloading the installer manually.
+<TabItem value="arch">
+
+1. Download [LLVM 10.0.0 pre-built binary for Ubuntu 18.04](https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz).
+2. Update the environment variables `TAICHI_CMAKE_ARGS` and `PATH`:
+
+  ```shell
+  export TAICHI_CMAKE_ARGS="-DCMAKE_CXX_COMPILER=<PATH_TO_LLVM_FOLDER>/bin/clang++ $TAICHI_CMAKE_ARGS"
+
+  export PATH=<PATH_TO_LLVM_FOLDER>/bin:$PATH
+  ```
+
+  :::tip NOTE
+
+  Some Linux distributions may require additional packages to build Taichi. Keep an eye on the output of CMake when building from source.
+
+  :::
+
+</TabItem>
+
+<TabItem value="others">
+
+Search [this site](https://pkgs.org/) for a Clang version that Taichi supports.
+
+:::tip NOTE
+
+Some Linux distributions may require additional packages to build Taichi. Keep an eye on the output of CMake when building from source.
 
-:::note
-If you are using a machine with an earlier CUDA version and/or old generation GPUs. We suggest to consult the [Compatibility Document](https://docs.nvidia.com/deploy/cuda-compatibility/) and the [CUDA Installation Guide](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html) first.
 :::
 
-### Setting up Vulkan (optional)
+</TabItem>
+
+</Tabs>
+
+### Install LLVM
+
+#### Install pre-built, customized LLVM binaries
+
+We provide pre-built, customized LLVM binaries. For now, Taichi supports LLVM 10.0.0 only. 
+
+1. Download and install customized binaries from the following list per your system environment:
+
+- [LLVM 10.0.0 for Linux](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_linux_patch2/taichi-llvm-10.0.0-linux.zip)
+- [LLVM 10.0.0 for macOS (without M1 chip)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-macos.zip)
+- [LLVM 10.0.0 for macOS (with M1 chip)](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10_m1/llvm-10.0.0-m1.zip)
+- [LLVM 10.0.0 for Windows MSVC 2019](https://github.com/taichi-dev/taichi_assets/releases/download/llvm10/taichi-llvm-10.0.0-msvc2019.zip)
+
+2. Add `$LLVM_FOLDER/bin` to `PATH`:
+
+<Tabs
+  defaultValue="linux"
+  values={[
+    {label: 'Linux & macOS', value: 'linux'},
+    {label: 'Windows', value: 'windows'},
+  ]}>
+
+<TabItem value="linux">
+
+1. Add LLVM to your PATH variable:
+   ```
+   export PATH=<PATH_TO_LLVM_FOLDER>/bin:$PATH
+   ```
+2. Update your path for the remainder of the session:
+
+   ```shell
+   source ~/.bashrc
+   ```
+
+</TabItem>
+
+<TabItem value="windows">
+
+Skip this section because we provide scripts that will install LLVM for you. See the [Build Taichi from source](#build-taichi-from-source) for more information. 
+
+</TabItem>
+
+</Tabs>
+
+
+<details>
+
+<summary><font color="#006284"><h2>Build LLVM 10.0.0 from source</h2></font></summary>
```

Comment:
Should be <h4> to be at the same level of `Install pre-built, customized LLVM binaries`?

Reply:
Yes. It ought to be placed at the same level with "Install pre-built, customized LLVM binaries".